packages/headless/src/controllers/facets/facet-search/facet-search.ts

```diff
@@ -77,6 +82,18 @@ export function buildGenericFacetSearch<T extends FacetSearchState>(
       dispatch(executeSearch(logFacetSelect({facetId, facetValue})));
     },
 
+    /** Resets the query and empties the values. */
+    clear() {
+      dispatch(
+        updateFacetSearch({
+          facetId,
+          query: defaultFacetSearchOptions.query,
+          numberOfValues: initialNumberOfValues,
+        })
+      );
+      dispatch(clearFacetSearchResults({facetId}));
```

Comment:
The `clearFacetSearchResults` action seems too granular and not useful on it's own. Is there a use-case where someone would want to dispatch it and not dispatch `updateFacetSearch` with the default query?


I think it be more useful to either:
A) Pass `initialNumberOfValues` as a parameter to `clearFacetSearchResults` (no need for the default query), or
B) Refactor to store the `initialNumberOfValues` in state so passing the parameter is not needed.

I think (B) is the better dev experience, but will take some refactoring. Depending on how much, (A) would be a pragmatic compromise,

Reply:
> Is there a use-case where someone would want to dispatch it and not dispatch updateFacetSearch with the default query

In atomic we clear the query & empty the results when the user "blurs out". Perhaps another dev would want to keep the query in the state, so when a new search is executed, it is with the same query. Other conditions can affect the facet results so keeping the results in state without triggering a new search can be problematic.

Reply:
That's a good point, the facet search results could become stale if someone interacts with other parts of the page.
Rather than needing developers to manually dispatch an action to clear the facet search results though, couldn't headless handle this behind the scenes, clearing the results on `executeSearch.fulfilled`? This feels like the better approach?


What I want to avoid is people using Headless for hide/show situations that are better handled with CSS.

If on blur someone wants to keep the query, they should hide the dropdown with CSS. If no search is done, then on refocus, the search results are still available. This does two things:

1. It saves a query.
2. It allows rendering the values instantly, without waiting on the server.


If on blur, someone wants to reset the facet search query and results, then they can call the new controller method you are adding. However, the action powering the interaction should also be cohesive so that developers are not dependent on the controller.

Reply:
Leveraging `executeSearch.fulfilled` is a good idea. And I'll make the action stand on its own